homeassistant/components/mqtt/client.py

```diff
@@ -178,7 +178,7 @@ async def async_subscribe(
     | AsyncDeprecatedMessageCallbackType,
     qos: int = DEFAULT_QOS,
     encoding: str | None = DEFAULT_ENCODING,
-):
+) -> Callable[[], None]:
```

Comment:
I think it makes sense to import and use `CALLBACK_TYPE` here.
```suggestion
) -> CALLBACK_TYPE:
```

side note: the changes to `homeassistant/components/mqtt/client.py` could have been in a separate PR for general type improvement.